vowpalwabbit/cb_adf.cc

```diff
@@ -506,9 +506,8 @@ base_learner* cb_adf_setup(options_i& options, vw& all)
     return nullptr;
 
   // Ensure serialization of this option in all cases.
-  if (!options.was_supplied("cb_type"))
+  if (options.ensure_default_dependency("cb_type", type_string))
   {
-    options.insert("cb_type", type_string);
     options.add_and_parse(new_options);
```

Comment:
why don't we avoid parsing if we can modify everything in memory. We would just need a separate api that skips all the communication via string.

Reply:
Yes, technically type_string could be set and we could avoid the parse and we can then make `ensure_default_dependency` be a void func